### PR TITLE
Feature/azure verifier

### PR DESCRIPTION
**Pull request by Jay-Motka-216**
### Added Azure SSO Token Verification

Added a new class for Azure SSO Token Verification

### Pull-Request Checklist

- [ ✅ ] The code is up-to-date with the `main` branch
- [ ✅ ] `yarn lint` passes with this change
- [ ✅ ] `yarn test` passes with this change
- [ ] This pull request links relevant issues as `Fixes #0000`
- [ ] There are new or updated unit tests validating the change
- [ ✅ ] Documentation has been updated to reflect this change
- [ ✅ ] The new commits follow conventions outlined in the [conventional commit spec](https://www.conventionalcommits.org/en/v1.0.0/)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,6 +15,7 @@
     "type-check": "tsc --noEmit",
     "test": "yarn lint && yarn type-check",
     "test:unit": "mocha -r ts-node/register --timeout 999999 ./test/**/*test.ts",
+    "test:single": "mocha -r ts-node/register --timeout 999999",
```
Comment (KillDozerX2) [inline package.json:18]: What are we testing here?

File: src/jwt.ts [hunk 2]
```diff
@@ -4,11 +4,14 @@ import jwtDecode from "jwt-decode";
 
 export interface PublicKey {
   kid: string;
-  alg: string;
+  alg?: string;
```
Comment (KillDozerX2) [inline src/jwt.ts:7]: Algorithm must be always present.

File: src/jwt.ts [hunk 3]
```diff
@@ -4,11 +4,14 @@ import jwtDecode from "jwt-decode";
...
   kty: string;
   e: string;
   n: string;
   use: string;
+  x5c?: string[];
+  x5t?: string;
+  issuer?: string;
```
Comment (KillDozerX2) [inline src/jwt.ts:14]: Do we need these things in the Public Key or is it just Azure AD Specific.

File: src/jwt.ts [hunk 4]
```diff
@@ -126,7 +129,8 @@ export class Verifier<T extends BaseClaims> {
 
     const joseKey = await jose.JWK.asKey(myPublicKey);
 
-    const verifiedToken = await jose.JWS.createVerify(joseKey).verify(token);
+    const verifier = jose.JWS.createVerify([joseKey]);
+    const verifiedToken = await verifier.verify(token);
```
Comment (KillDozerX2) [inline src/jwt.ts:133]: Why change it to two steps? For debugging or readability?

File: test/api.test.ts [hunk 5]
```diff
@@ -16,6 +16,12 @@ describe("API Test Suite", () => {
     },
     "test-key"
   ).compact();
+  const azureToken = create(
+    {
+      email: "admin@distinction.dev",
+    },
+    "test-key"
+  ).compact();
```
Comment (KillDozerX2) [inline test/api.test.ts:24]: Why do we need this in API Tests?

File: test/azure.test.ts [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+//   exp: Math.floor((new Date().getTime() + 60 * 60 * 4) / 1000),
+//   iat: Math.floor(Date.now() / 1000),
+//   sub: "test",
+// });
+// const key = {
+//   kty: "RSA",
+//   use: "sig",
+//   kid: "-KI3Q9nNR7bRofxmeZoXqbHZGew",
+//   x5t: "-KI3Q9nNR7bRofxmeZoXqbHZGew",
+//   n: "tJL6Wr2JUsxLyNezPQh1J6zn6wSoDAhgRYSDkaMuEHy75VikiB8wg25WuR96gdMpookdlRvh7SnRvtjQN9b5m4zJCMpSRcJ5DuXl4mcd7Cg3Zp1C5-JmMq8J7m7OS9HpUQbA1yhtCHqP7XA4UnQI28J-TnGiAa3viPLlq0663Cq6hQw7jYo5yNjdJcV5-FS-xNV7UHR4zAMRruMUHxte1IZJzbJmxjKoEjJwDTtcd6DkI3yrkmYt8GdQmu0YBHTJSZiz-M10CY3LbvLzf-tbBNKQ_gfnGGKF7MvRCmPA_YF_APynrIG7p4vPDRXhpG3_CIt317NyvGoIwiv0At83kQ",
```
Comment (KillDozerX2) [inline test/azure.test.ts:18]: Remove unnecessary code please?

File: src/jwt.ts (hunk 4 above)
Comment (Jay-Motka-216) [reply src/jwt.ts:133]: For readability

File: test/api.test.ts (hunk 5 above)
Comment (Jay-Motka-216) [reply test/api.test.ts:24]: We do not need it. Will remove it

File: test/azure.test.ts (hunk 6 above)
Comment (Jay-Motka-216) [reply test/azure.test.ts:18]: Sure, will remove that

File: package.json (hunk 1 above)
Comment (Jay-Motka-216) [reply package.json:18]: Just added that for single file testing and other options but will remove that as we do not need that now.

File: src/jwt.ts (hunk 3 above)
Comment (Jay-Motka-216) [reply src/jwt.ts:14]: probably don't need the issuer but there is x5t key present in public key, but I guess the test is working without it so I will remove the keys from interface

File: src/jwt.ts (hunk 2 above)
Comment (Jay-Motka-216) [reply src/jwt.ts:7]: Will remove the optional operator

File: test/api.test.ts (hunk 5 above)
Comment (KillDozerX2) [reply test/api.test.ts:24]: Thanks

File: test/azure.test.ts (hunk 6 above)
Comment (KillDozerX2) [reply test/azure.test.ts:18]: Thanks

File: src/jwt.ts (hunk 3 above)
Comment (KillDozerX2) [reply src/jwt.ts:14]: Cool

**Review by KillDozerX2 — DISMISSED** [review]

**Comment by KillDozerX2** [comment]
Good to go!

**Review by KillDozerX2 — APPROVED** [review]

**Pull request merged**